packages/office-ui-fabric-react/src/components/Pivot/Pivot.base.tsx

```diff
@@ -192,38 +178,64 @@ export class PivotBase extends BaseComponent<IPivotProps, IPivotState> {
     );
   };
 
+  private _selectedKey(): string | undefined {
+    if (this._isKeyValid(this.props.selectedKey)) {
+      return this.props.selectedKey!;
+    }
+    if (this._isKeyValid(this.state.selectedKey)) {
+      return this.state.selectedKey;
+    }
+    if (this._links.length) {
+      return this._links[0].itemKey!;
+    }
+  }
+
+  private _isItemSelected(itemKey: string | undefined): boolean {
+    return this._selectedKey() === itemKey;
+  }
+
+  private _selectedTabId(): string | undefined {
+    const selectedKey = this._selectedKey();
+    if (selectedKey) {
+      return this._keyToTabIds[selectedKey];
+    }
+    return undefined;
+  }
+
   /**
    * Gets the set of PivotLinks as arrary of IPivotItemProps
    * The set of Links is determined by child components of type PivotItem
    */
-  private _getPivotLinks(props: IPivotProps): IPivotItemProps[] {
+  private _updatePivotLinks(): void {
     const links: IPivotItemProps[] = [];
     this._keyToIndexMapping = {};
     this._keyToTabIds = {};
 
-    React.Children.map(props.children, (child: any, index: number) => {
-      if (typeof child === 'object' && child.type === PivotItemType) {
-        const pivotItem = child as PivotItem;
-        const itemKey = pivotItem.props.itemKey || index.toString();
-
-        links.push({
-          headerText: pivotItem.props.headerText || pivotItem.props.linkText,
-          headerButtonProps: pivotItem.props.headerButtonProps,
-          ariaLabel: pivotItem.props.ariaLabel,
-          itemKey: itemKey,
-          itemCount: pivotItem.props.itemCount,
-          itemIcon: pivotItem.props.itemIcon,
-          onRenderItemLink: pivotItem.props.onRenderItemLink,
-          keytipProps: pivotItem.props.keytipProps
-        });
-        this._keyToIndexMapping[itemKey] = index;
-        this._keyToTabIds[itemKey] = this._getTabId(itemKey, index);
-      } else {
-        warn('The children of a Pivot component must be of type PivotItem to be rendered.');
-      }
-    });
+    if (this.props && this.props.children) {
```

Comment:
Shouldn't `this.props` always be defined? Also, I'd slightly prefer something like this (gets rid of the extra level of nesting and a bunch of whitespace changes):
```
const children = this.props.children || [];
React.Children.map(children, ...)
```